packages/@aws-cdk/aws-apigatewayv2-integrations/lib/alb-integration.ts

```diff
@@ -0,0 +1,40 @@
+import { HttpMethod, IVpcLink } from '@aws-cdk/aws-apigatewayv2';
+import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
+import { HttpProxyPrivateIntegration } from './http-private';
+
+/**
+ * Properties to initialize `ALBPrivateIntegration`.
+ */
+export interface ALBPrivateIntegrationProps {
+  /**
+   * The listener to the application load balancer used for the integration
+   */
+  readonly listener: elbv2.ApplicationListener;
+
+  /**
+   * The vpc link to be used for the private integration
+   *
+   * @default - a new VpcLink is created
+   */
+  readonly vpcLink?: IVpcLink;
+
+  /**
+   * The HTTP method that must be used to invoke the underlying HTTP proxy.
+   * @default HttpMethod.ANY
+   */
+  readonly method?: HttpMethod;
+}
```

Comment:
I think you can just extend `HttpProxyPrivateIntegrationProps` and drop everything except for the `listener` property.

Reply:
We cannot directly extend the `HttpPrivateIntegrationProps` interface because of required `targetArn` prop. Created a `BasePrivateIntegrationProps` interface for the same.

Note: The `HttpServiceDiscoveryIntegration` integration does not extend this because of the required `vpcLink` prop.

Reply:
I'm not a fan of the `BasePrivateIntegrationProps`, especially because it needs to exported publicly. No value in exposing this to customers and increasing the surface area.

I think I know what's going on here. The root problem is that the integration between `HttpPrivateIntegration` and any of its subclasses is through the constructor.

How about instead changing this to be via `bind()`? Technically, each implementation of `IHttpRouteIntegration` should re-implement `bind()` suitable to it. This would look something like - 

```ts
export abstract class HttpPrivateIntegration implements IHttpRouteIntegration {
  protected configureVpcLink(bindOptions: HttpRouteIntegrationBindOptions, integrationOptions: HttpPrivateIntegrationOptions): VpcLink {
    // ...
    return options.route.httpApi.addVpcLink(...);
  }

  public abstract bind(options: HttpRouteIntegrationBindOptions);
}

export class HttpAlbIntegration extends HttpPrivateIntegration {
  constructor(private readonly props: HttpAlbIntegrationProps) {}

  public abstract bind(options: HttpRouteIntegrationBindOptions) {
    const vpclink = this.configureVpcLink(options, props);
    return {
      ...
    }
  }
}
```

With this form of structuring, it should be possible for `HttpAlbIntegrationProps` to extend `HttpPrivateIntegrationOptions` (formerly: `HttpPrivateIntegrationProps`).

An added benefit is that, each specific implementation has a lot more control over how it's bound. Any commonality like `PayloadFormatVersion` can be captured at the base class but can be overridden by a specific implementation if it wishes to.

Does this make sense?

Reply:
I'll go one step further and say, we shouldn't even export `HttpPrivateIntegration`.

A [limitation of JSII](https://github.com/aws/jsii/issues/2159) would mean that all sub-classes will need to be in the same file, but looking at the size of these classes, this might be ok.

Reply:
Updated my comment above

Reply:
Yes that makes sense and is better way to implement. Will make the appropriate changes.